sub-packages/bionemo-testing/tests/bionemo/testing/test_distributed.py

```diff
@@ -0,0 +1,115 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import torch
+import torch.distributed as dist
+import torch.multiprocessing.spawn
+from megatron.core import parallel_state
+
+from bionemo.testing.distributed import dist_environment
+
+
+REQUIRED_WORLD_SIZE = 2
+
+
+# TODO @sichu improve documentation
+#  arguments will be recognized as fixture if decorating the private function directly into test function,
+#  so we have to create a test function for each private function.
+def _test_all_reduce_sum(rank: int, world_size: int):
+    """Private test function for torch.distributed mean reduce."""
+    with dist_environment(rank=rank, world_size=world_size):
+        tensor = torch.tensor([rank + 1]).cuda(rank)
+        dist.all_reduce(tensor)
+        assert tensor.item() == world_size * (world_size + 1) / 2
```

Comment:
could you move this function definition inside `test_all_reduce_sum`, and just call it something like `all_reduce_sum`?

Reply:
ditto with the remaining functions in this pattern